src/control/BarrettHandKinematicSimulationPositionCommandExecutor.cpp

```diff
@@ -121,42 +133,45 @@ std::future<void>
 BarrettHandKinematicSimulationPositionCommandExecutor::execute(
     const Eigen::VectorXd& goalPositions)
 {
-  std::lock_guard<std::mutex> lockSpin(mMutex);
+  std::lock_guard<std::mutex> lock(mMutex);
 
-  if (mInExecution)
-    throw std::runtime_error("Another command in execution.");
+  if (mInProgress)
+    throw std::runtime_error("Another position command is in progress.");
 
   if (goalPositions.size() != 4)
   {
     std::stringstream message;
-    message << "GoalPositions must have 4 elements, but ["
+    message << "BarrettHand goal positions must have 4 elements, but ["
             << goalPositions.size() << "] given.";
     throw std::invalid_argument(message.str());
   }
 
   mPromise.reset(new std::promise<void>());
-  mProximalGoalPositions = goalPositions.head<3>();
-  mSpreadGoalPosition = goalPositions.row(3);
-  mInExecution = true;
+
+  Eigen::VectorXd proximalGoalPositions = goalPositions.head<3>();
+  Eigen::VectorXd spreadGoalPosition = goalPositions.row(3);
   mFingerFutures.clear();
 
-  mFingerFutures.reserve(kNumPositionExecutor + kNumSpreadExecutor);
-  for (std::size_t i = 0; i < kNumPositionExecutor; ++i)
+  mFingerFutures.reserve(kNumPositionExecutors + kNumSpreadExecutor);
+  for (std::size_t i = 0; i < kNumPositionExecutors; ++i)
     mFingerFutures.emplace_back(
-        mPositionCommandExecutors[i]->execute(mProximalGoalPositions.row(i)));
+        mPositionCommandExecutors[i]->execute(proximalGoalPositions.row(i)));
```

Comment:
Nit: Since `proximalGoalPositions` is a vector, we might want to use `[i]` instead of `row(i)`.

Reply:
The `PositionCommandExecutor::execute` interface requires this to be an `Eigen::VectorXd`, not a double.

Reply:
If so, this line makes sense. :smile: